lib/dockerregistry/inventory_test.go

```diff
@@ -2458,6 +2458,80 @@ func TestPromotion(t *testing.T) {
 	}
 }
 
+func TestExecRequests(t *testing.T) {
+	sc := reg.SyncContext{}
+
+	srcRegName := reg.RegistryName("gcr.io/foo")
+	destRegName := reg.RegistryName("gcr.io/bar")
+	destRegName2 := reg.RegistryName("gcr.io/cat")
+	destRC := reg.RegistryContext{
+		Name:           destRegName,
+		ServiceAccount: "robot",
+	}
+	destRC2 := reg.RegistryContext{
+		Name:           destRegName2,
+		ServiceAccount: "robot",
+	}
+	srcRC := reg.RegistryContext{
+		Name:           srcRegName,
+		ServiceAccount: "robot",
+		Src:            true,
+	}
+	registries := []reg.RegistryContext{destRC, srcRC, destRC2}
+
+	nopStream := func(
+		srcRegistry reg.RegistryName,
+		srcImageName reg.ImageName,
+		rc reg.RegistryContext,
+		destImageName reg.ImageName,
+		digest reg.Digest,
+		tag reg.Tag,
+		tp reg.TagOp) stream.Producer {
+
+		// We don't even need a stream producer, because we are not creating
+		// subprocesses that generate JSON or any other output; the vanilla
+		// "mkReq" in Promote() already stores all the info we need to check.
+		return nil
+	}
+
+	successEdges, _ := reg.ToPromotionEdges([]reg.Manifest{reg.Manifest{}})
+	populateRequestsFakeSuccess := reg.MKPopulateRequestsForPromotionEdges(
+		successEdges,
+		nopStream)
+
+	errorEdges, _ := reg.ToPromotionEdges([]reg.Manifest{reg.Manifest{
+		Registries: registries,
+		Images: []reg.Image{
+			{
+				ImageName: "a",
+				Dmap: reg.DigestTags{
+					"sha256:000": {"0.9"}}},
+			{
+				ImageName: "b",
+				Dmap: reg.DigestTags{
+					"sha256:111": {"0.9"}}}},
+		SrcRegistry: &srcRC}})
+	populateRequestsFakeError := reg.MKPopulateRequestsForPromotionEdges(
+		errorEdges,
+		nopStream)
+
+	captured := make(reg.CapturedRequests)
+	processRequestFake := reg.MkRequestCapturer(&captured)
```

Comment:
Your `processRequestFake` function here needs to (always) populate the requestResults channel with an error. This will force your new ExecRequests function to always set the boolean you added to `true`, enabling you to test it here. Likewise your your sucessEdges case, you would have a `processRequestFakeSucceed` function that always processes the requests by populating requestResults with non-errors (probably just `reqRes.Context`).